src/datasets/iterable_dataset.py

```diff
@@ -568,16 +585,21 @@ def map(
                 as positional arguments. If `None`, a dict mapping to all formatted columns is passed as one argument.
             batched (:obj:`bool`, default `False`): Provide batch of examples to `function`.
             batch_size (:obj:`int`, optional, default ``1000``): Number of examples per batch provided to `function` if `batched=True`.
-            remove_columns (`Optional[List[str]]`, defaults to `None`): Remove a selection of columns while doing the mapping.
+            remove_columns (`Optional[Union[bool,List[str]]]`, defaults to `None`): Remove a selection of columns while doing the mapping.
                 Columns will be removed before updating the examples with the output of `function`, i.e. if `function` is adding
                 columns with names in `remove_columns`, these columns will be kept.
+                Argument behaviour according to typing:
+                    - None: No column is removed.
+                    - bool: Flag determining whether all columns are removed or not. Columns from the output of `function` are preserved.
+                    - str: Single column removed. Columns from the output of `function` are preserved.
+                    - List[str]: List of columns removed. Columns from the output of `function` are preserved.
         """
         if isinstance(input_columns, str):
             input_columns = [input_columns]
         if isinstance(remove_columns, str):
             remove_columns = [remove_columns]
         if function is None:
-            function = lambda x: x  # noqa: E731
+            function = lambda x: {}  # noqa: E731
```

Comment:
That's a direct consequence of the fix, this was an inplace operation, which means that if you remove a column, it would get removed from both `input` and `processed_input` which causes the test to work, but in fact given the API we have for `arrow_dataset` this should have returned an unchanged dataset.